thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js

```diff
@@ -86,17 +87,24 @@ export default Ember.Component.extend({
      * @method filterByEvent
      * @param {Object} clickedBlock - selected filter block object
      */
-    filterByEvent(clickedBlock) {
+    selectEventType(clickedBlock) {
       // Hide all other blocks when one is clicked
-      let filterBlocks = this.get('filterBlocks');
+      let filterBlocks = this.get('config');
+
       filterBlocks.forEach(block => {
         Ember.set(block, 'isHidden', true);
       });
 
       // Note: toggleProperty will not be able to find 'filterBlocks', as it is not an observed property
       // Show clickedBlock
       Ember.set(clickedBlock, 'isHidden', !clickedBlock.isHidden);
-      this.attrs.onTabChange(clickedBlock.eventType);
+
+      const { entities, onSelect } = this.getProperties('entities', 'onSelect');
+      if (onSelect != null) {
+        const urns = Object.keys(entities).filter(urn => entities[urn].type == 'event')
```

Comment:
can we just do `entities[urn].type == 'event' && entities[urn].eventType == clickedBlock.eventType` in one pass?

Reply:
Yep! Thanks for the comment. It's fixed.